datalad/support/gitrepo.py

```diff
@@ -3363,34 +3363,36 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
         # _status=None, we should be able to avoid this, because
         # status should have the full info already
         # looks for contained repositories
-        to_add_submodules = [sm for sm, sm_props in iteritems(
-            self.get_content_info(
-                # get content info for any untracked directory
-                [f.relative_to(self.pathobj) for f, props in iteritems(status)
-                 if props.get('state', None) == 'untracked' and
-                 props.get('type', None) == 'directory'],
```

Comment:
Aside from pulling this argument out as a variable, this diff boils down to indenting the current code under a new conditional.